src/codec.ts

```diff
@@ -84,73 +84,67 @@ const TYPE = Symbol();
  *
  * @returns {array}
  */
-function Struct() {
-  const struct = [];
+export class Struct extends Array {
+  constructor() {
+    super();
+    this[TYPE] = Struct.TYPE;
+    Object.defineProperty(this, 'toJSON', {
+      enumerable: false,
+      value: codec.generateToJSONFromRow(this)
+    });
+  }
 
-  struct[TYPE] = Struct.TYPE;
+  /**
+   * Use this to assign/check the type when dealing with structs.
+   *
+   * @private
+   */
+  static TYPE = 'struct';
+
+  /**
+   * Converts an array of objects to a struct array.
+   *
+   * @private
+   *
+   * @param {object[]} arr Struct array.
+   * @return {Struct}
+   */
+  static fromArray(arr) {
+    const struct = new Struct();
+    struct.push.apply(struct, arr);
+    return struct;
+  }
 
-  Object.defineProperty(struct, 'toJSON', {
-    enumerable: false,
-    value: codec.generateToJSONFromRow(struct),
-  });
+  /**
+   * Converts a JSON object to a struct array.
+   *
+   * @private
+   *
+   * @param {object} json Struct JSON.
+   * @return {Struct}
+   */
+  static fromJSON(json: {}) {
+    const struct = new Struct();
+    Object.keys(json || {}).forEach(name => {
+      const value = json[name];
+      struct.push({name, value});
+    });
+    return struct;
+  }
 
-  return struct;
+  /**
+   * Checks to see if the provided object is a Struct.
+   *
+   * @private
+   *
+   * @param {*} thing The object to check.
+   * @returns {boolean}
+   */
+  static isStruct(thing: {}) {
```

Comment:
Good opportunity for:
```suggestion
  static isStruct(thing: {}) : thing is Struct {
```